common/commands/src/validator/mixnet/query/query_all_gateways.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: Apache-2.0
+
+use clap::Parser;
+use comfy_table::Table;
+use log::error;
+
+use crate::context::QueryClient;
+use crate::utils::pretty_cosmwas_coin;
+
+#[derive(Debug, Parser)]
+pub struct Args {
+    #[clap(value_parser)]
+    #[clap(help = "Optionally, the gateway to display")]
+    pub identity_key: Option<String>,
+}
+
+pub async fn query(args: Args, client: &QueryClient) {
+    match client.get_gateways_paged(None, None).await {
```

Comment:
This is not the query you want to use. If there are more gateways in the system than the smart contract page limit, it will not show anything more than that. What you want to use is `get_all_nymd_gateways`